x-pack/plugins/maps/server/content_management/maps_storage.ts

```diff
@@ -4,338 +4,32 @@
-      throw Boom.badRequest(`Invalid options. ${optionsError.message}`);
-    }
-
-    // Save data in DB
-    const soClient = await savedObjectClientFromRequest(ctx);
-    const partialSavedObject = await soClient.update<MapAttributes>(
-      SO_TYPE,
-      id,
-      dataToLatest,
-      optionsToLatest
-    );
-
-    // Validate DB response and DOWN transform to the request version
-    const { value, error: resultError } = transforms.update.out.result.down<
-      MapUpdateOut,
-      MapUpdateOut
-    >({
-      item: savedObjectToMapItem(partialSavedObject, true),
-    });
-
-    if (resultError) {
-      throw Boom.badRequest(`Invalid response. ${resultError.message}`);
-    }
-
-    return value;
-  }
-
-  async delete(ctx: StorageContext, id: string): Promise<MapDeleteOut> {
-    const soClient = await savedObjectClientFromRequest(ctx);
-    await soClient.delete(SO_TYPE, id);
-    return { success: true };
-  }
-
-  async search(
-    ctx: StorageContext,
-    query: SearchQuery,
-    options: MapSearchOptions = {}
-  ): Promise<MapSearchOut> {
-    const {
-      utils: { getTransforms },
-    } = ctx;
-    const transforms = getTransforms(cmServicesDefinition);
-    const soClient = await savedObjectClientFromRequest(ctx);
-
-    // Validate and UP transform the options
-    const { value: optionsToLatest, error: optionsError } = transforms.search.in.options.up<
-      MapSearchOptions,
-      MapSearchOptions
-    >(options);
-    if (optionsError) {
-      throw Boom.badRequest(`Invalid payload. ${optionsError.message}`);
-    }
-    const { onlyTitle = false } = optionsToLatest;
-
-    const { included, excluded } = query.tags ?? {};
-    const hasReference: SavedObjectsFindOptions['hasReference'] = included
-      ? included.map((id) => ({
-          id,
-          type: 'tag',
-        }))
-      : undefined;
-
-    const hasNoReference: SavedObjectsFindOptions['hasNoReference'] = excluded
-      ? excluded.map((id) => ({
-          id,
-          type: 'tag',
-        }))
-      : undefined;
-
-    const soQuery: SavedObjectsFindOptions = {
-      type: CONTENT_ID,
-      search: query.text,
-      perPage: query.limit,
-      page: query.cursor ? +query.cursor : undefined,
-      defaultSearchOperator: 'AND',
-      searchFields: onlyTitle ? ['title'] : ['title^3', 'description'],
-      fields: ['description', 'title'],
-      hasReference,
-      hasNoReference,
-    };
-
-    // Execute the query in the DB
-    const response = await soClient.find<MapAttributes>(soQuery);
+};
 
-    // Validate the response and DOWN transform to the request version
-    const { value, error: resultError } = transforms.search.out.result.down<
-      MapSearchOut,
-      MapSearchOut
-    >({
-      hits: response.saved_objects.map((so) => savedObjectToMapItem(so, false)),
-      pagination: {
-        total: response.total,
-      },
+export class MapsStorage extends SOContentStorage<MapCrudTypes> {
+  constructor() {
```

Comment:
This is the result. Expressing your SO / content mgmt api integration in a very concise number of lines of code. Easier to see if you look at the file view